crates/task-impls/src/consensus/view_change.rs

```diff
@@ -31,12 +26,9 @@ use crate::{
 /// # Errors
 /// Returns an [`anyhow::Error`] when the new view is not greater than the current view.
 #[allow(clippy::too_many_arguments)]
-pub(crate) async fn update_view<TYPES: NodeType, I: NodeImplementation<TYPES>>(
-    public_key: TYPES::SignatureKey,
+pub(crate) async fn update_view<TYPES: NodeType>(
```

Comment:
Not sure if this still violates `too_many_arguments` with the three parameters removed.